File: Store/Entity/models/Product.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.AspNetCore.Http;

namespace Entity.models
{
    public class Product : Entity
    {
        public float Price { get; set; }

        public string ShortDescription { get; set; }

        public string Description { get; set; }

        public Category Category { get; set; }

        public virtual ICollection<Review> Reviews { get; set; }

        [NotMapped]
        public IFormFile UploadedImage { get; set; }

        public string ImageUrl { get; set; }

        public virtual ICollection<Specification> Specifications { get; set; }
    }
}

File: Store/ClientApp/src/app/models/product.ts
export interface Product {
  price:            number;
  shortDescription: string;
  description:      string;
  category:         Category;
  reviews:          null;
  uploadedImage:    null;
  imageUrl:         null;
  specifications:   Specification[];
  name:             string;
  id:               string;
}

export interface Category {
  order: number;
  name:  string;
  id:    string;
}

export interface Specification {
  value: string;
  name:  string;
  id:    string;
}
